internal/tiltfile/tiltfile.go

```diff
@@ -382,17 +382,27 @@ func Load(ctx context.Context, filename string) (*Tiltfile, error) {
 // a manifest representing the global yaml
 func (t Tiltfile) GetManifestConfigsAndGlobalYAML(names ...string) ([]model.Manifest, model.YAMLManifest, error) {
 	var manifests []model.Manifest
+
+	globalYAML := model.NewYAMLManifest(model.GlobalYAMLManifestName, t.globalYAMLStr, t.globalYAMLDeps)
```

Comment:
Actually do we need `WithDependencies` if we move this line down below the for loop and pass in the fully realized list of config files to the `NewYAMLManifest` constructor?

Since this is a new struct my instinct is to make all the fields private, which I haven't done, and try to have as many fields initialized at creation time as possible rather than have builders for all of the fields.

Reply:
nah we need `WithDependencies` (or `WithConfigFiles` as the case may be) to modify the computed _resource_ manifests so all treat globalYAML dependencies as their own dependencies. Unless I'm missing something...?

Reply:
Oh- I misunderstood, for some reason I thought you were modifying the global YAML manifest. Carry on!